l10n_br_sale/models/sale.py

```diff
@@ -0,0 +1,376 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+# Copyright (C) 2009  Renato Lima - Akretion
+# Copyright (C) 2012  Raphaël Valyi - Akretion
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+###############################################################################
+
+from openerp import models, fields, api
+from openerp.addons import decimal_precision as dp
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.one
+    @api.depends('order_line.price_unit', 'order_line.tax_id',
+                 'order_line.discount', 'order_line.product_uom_qty')
+    def _amount_all_wrapper(self):
+        """ Wrapper because of direct method passing
+        as parameter for function fields """
+        return self._amount_all()
+
+    @api.one
```

Comment:
forget these remarks: we are overriding super methods in these cases so we cannot change the api to multi. Furthermore, the purpose of using non deprecated new API is to have the closest code possible between 8 and 9. But in any case that will not be possible here as v9 comes with a cleaned new API in the sale order core.